backend/src/main/java/saci/infrastructure/LevelRepository.java

```diff
@@ -11,9 +12,22 @@ public interface LevelRepository extends JpaRepository<Level, Long> {
     Optional<Level> findByName(String name);
 
     @Query(
-            "SELECT COUNT(*) FROM Level l WHERE l.roleId = :roleId AND (l.minCoefficient < :maxCoefficient AND l.maxCoefficient > :minCoefficient)")
+            "SELECT COUNT(l) FROM Level l WHERE l.roleId = :roleId AND "
+                    + "(l.minCoefficient < :maxCoefficient AND l.maxCoefficient > :minCoefficient)")
     int overlappingLevelsCounter(
             @Param("roleId") Long roleId,
-            @Param("minCoefficient") Integer minCoefficient,
-            @Param("maxCoefficient") Integer maxCoefficient);
+            @Param("minCoefficient") Double minCoefficient,
+            @Param("maxCoefficient") Double maxCoefficient);
+
+    @Query(
+            "SELECT l FROM Level l WHERE l.roleId = :roleId AND "
+                    + ":score BETWEEN l.minCoefficient AND l.maxCoefficient")
+    Optional<Level> findByRoleIdAndScoreBetweenCoefficients(
+            @Param("roleId") Long roleId, @Param("score") double score);
+
+    @Query(
+            "SELECT l FROM Level l WHERE l.roleId = :roleId AND l.minCoefficient > :score "
+                    + "ORDER BY l.minCoefficient ASC")
+    Optional<Level> findFirstByRoleIdAndMinCoefficientGreaterThanOrderByMinCoefficientAsc(
```

Comment:
The objective here was to shorten the method name, please give one more suitable like you gave in the service "findNextLevelBasedOfScore".